app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt

```diff
@@ -127,9 +129,13 @@ class OrdersUnderUserFragment : Fragment(), BottomIconDoubleClick {
         super.onViewCreated(view, savedInstanceState)
 
         val recyclerViewClickListener = object : OrdersPagedListAdapter.OrderClickListener {
-            override fun onClick(eventID: Long, orderIdentifier: String, orderId: Long) {
-                findNavController(rootView).navigate(OrdersUnderUserFragmentDirections
-                    .actionOrderUserToOrderDetails(eventID, orderIdentifier, orderId))
+            override fun onClick(eventID: Long, orderIdentifier: String, order: Order) {
+                if (order.status == ORDER_STATUS_PENDING)
+                    findNavController(rootView).navigate(OrdersUnderUserFragmentDirections
+                        .actionOrderUserToPaypalPayment(order.identifier.nullToEmpty()))
+                else
+                    findNavController(rootView).navigate(OrdersUnderUserFragmentDirections
+                        .actionOrderUserToOrderDetails(eventID, orderIdentifier, order.id))
```

Comment:
Open it when order status is completed or placed else show snack bar. And for pending, a dialog should display for payment confirmation.